External/FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -636,7 +686,63 @@ namespace FEXCore::Context {
     Thread->LookupCache->AddBlockMapping(Address, Ptr, Start, Length);
   }
 
+  void Context::SMCMarkReadOnly(uint64_t Start, uint64_t Length) {
+    const auto Base = Start & FHU::FEX_PAGE_MASK;
+    const auto Top = FEXCore::AlignUp(Start + Length, FHU::FEX_PAGE_SIZE);
+
+    {
+      FHU::ScopedSignalMaskWithSharedMutex lk(MemoryEntryMutex, true);
+
+      // find the first Mapping at or after the Range ends, or ::end()
+      // Top is the first value after the end mapping
```

Comment:
```suggestion
      // Top points to the address after the end of the range
```

Reply:
Uh, I don't think it is repeating, it's making a note that `Top` is not inclusive. As it's very easy to have bugs between `upper_bound` and `lower_bound`, it is there to help you, so you don't have to scan the code for it.

Reply:
Ah, I misread what you're trying to say in the second line. But even with my new understanding, I think that line adds more confusion than just making the reader look at the definition of "Top" since it's not clear *why* you're mentioning this. (Also note that the description of `Top` is incorrect, since it's the start of the first page following the range rather than the range end.)

> As it's very easy to have bugs between upper_bound and lower_bound, it is there to help you, so you don't have to scan the code for it.

No, this is over-optimizing for proofreading at the expense of having a clear description of the high-level behavior. IMO it's sufficient to explain what result is *desired*, and it's passable to require a quick look upwards when checking that the `lower_bound` actually computes the described result.

How about `Find the first mapping following the input range, or ::end`? For extra clarity, `MemoryMaps.lower_bound(Start + Size)` could be used instead.